### ISSUE TITLE
Restrict task from being handled by a single worker

### ISSUE THREAD

**Issue by Eagllus** [Koed00/django-q#301]
@jordanmkoncz I created a new issue for your question. This is completely different question then that @mrname asked.

@Eagllus apologies if this is explained in the docs, I tried to find it but wasn't able to. How exactly would you "abort the task"? Do you simply mean have the task return early before doing whatever work it would normally do?

I tried using the chains functionality for the problem I had, but unfortunately it's not quite what I need. The tasks for which I'm trying to limit parallel execution are requested by end-users which means they can be requested at any time and in any quantity; i.e. sometimes 10 of them could be requested within a 2 minute period, which would take up all of my workers.

I'm trying to find a solution in which most tasks just get added to the queue at any time and are processed normally (i.e. they just get picked up by any available worker and will complete quite quickly), but a special type of task can only be running on a single worker at any point in time. That is, if I have 5 workers and one of these special tasks is already being processed by a worker and there are 10 other special tasks sitting in the queue ready to be processed then the other 10 will just remain in the queue (with the other 4 workers either doing nothing, or processing normal tasks); only after the special task currently being processed has completed will one of the 10 in the queue be picked up by a worker.

It seems like django-q does not currently provide any way to achieve this, so I'm wondering whether I could use database locks like you suggested as a way to achieve this. But the thing is, if one of my special tasks gets picked up by a worker and then determines that a lock is already present (i.e. another special task is already being processed), I need the task to just be returned to the queue rather than just doing a return to abort early, and then being considered 'completed' even though the work for that task was not actually completed.

Do you have any suggestions for how I could achieve the functionality I'm looking for?

**Comment by Eagllus** [comment]
By default this is not something Django-Q can solve for you.
There are ways to achieve this so I'll try and explain 2 of them.

1. By creating a different tiny application that only does 1 things.
It will handle the special tasks from a different queue that is filled by the main application.
This is quite easy to setup because Django ORM is able to handle multiple database.

2. Using multiple clusters.
By running `python manage.py qcluster --settings <settings_2>` you can change the settings django-q is using. This will allow you to have different clusters that could pickup tasks from different queues.
If your using the ORM queue this means you have to configure a different queue for the `special cluster`. I would recommend using Redis for the special queue.

In your main settings you would have something like.
```
...
Q_CLUSTER = {
    'name': '<application_name>',
    'workers': 4,
    'orm': 'default',
}
...
```
calling it by `python manage.py qcluster`


While the special cluster settings will look something like.
```
...
Q_CLUSTER = {
    'name': '<application_name>',
    'workers': 1,
    'orm': 'redis',
}
...
```
calling it by `python manage.py qcluster --settings '<application_name>.settings.special_cluster'`
